ts/src/header-validator/validate-json.ts

```diff
@@ -1165,6 +1215,9 @@ export type Source = CommonDebug &
     triggerDataMatching: TriggerDataMatching
 
     eventLevelEpsilon: number
+    attributionScopes: Set<string> | null
```

Comment:
Nice catch, I believe currently all attribution scope fields are nullable (in the explainer). will update the spec field to match that.

Reply:
I don't understand what it means for this field to be `null` rather than the empty set. If a source omits the scopes field, shouldn't it have an empty set?

Reply:
- null : matches all sources, with or without scopes.
- empty set: matches to none.

As discussed in the other comments, we can't reserve empty set to select sources without scopes as that will be equivalent to introduce a new scope.

Reply:
I understand null vs empty for triggers, but what about for sources?

A source either has 0 scopes or it has multiple scopes. What does null mean for a source instead of 0?